### ISSUE TITLE
ICE in LLVM: "Unexpected illegal type" at llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp:978

### ISSUE THREAD

**Issue by Vsevolod-Livinskij** [ispc/ispc#1844]
ISPC fails with an assertion. The same error exists for avx512skx-i16x32.

>$ ispc -O2 --target=avx512skx-i8x64 -o func.o func.ispc
ispc: llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp:978: void {anonymous}::SelectionDAGLegalize::LegalizeOp(llvm::SDNode*): Assertion `(TLI.getTypeAction(*DAG.getContext(), Op.getValueType()) == TargetLowering::TypeLegal || Op.getOpcode() == ISD::TargetConstant || Op.getOpcode() == ISD::Register) && "Unexpected illegal type!"' failed.
ispc/src/main.cpp(372): FATAL ERROR: Unhandled signal sent to process; terminating.

```c++
extern uniform unsigned int64 var_6;
extern uniform unsigned int16 var_11;
extern uniform bool arr_201 [1] [15] [20];
extern uniform unsigned int8 arr_202 [1] [15] [20];
extern uniform int64 arr_210 [1] [15] [20] [26] [11];

export void test() {
    foreach(i_53 = var_6...0L)
        for (uniform unsigned int16 i_54 = 0L; i_54 < 14L; i_54 += 4L)
            for (uniform unsigned int8 i_55 = 0; i_55 < 19; i_55 += 2)
                for (uniform unsigned int8 i_57 = 0L; i_57 < 25L; i_57 += 4)
                    for (uniform unsigned int8 i_58 = 0; i_58 < extract(((varying int32)(varying unsigned int16)var_11) ^ arr_201 [i_53] [i_54] [i_55], 0U) - 120L; i_58 += 4L) 
                        if (arr_201 [i_53] [i_54] [i_55])
                            arr_210 [i_53] [i_54] [i_55] [i_57] [i_58] = -arr_202 [i_53] [i_54] [i_55];
}
```

ISPC version:
1.15.0dev (build commit 0988655a3ef588c3 @ 20200731, LLVM 10.0.1)

**Comment by dbabokin** [comment]
Test fails with with LLVM 10.0.1 (the one used for ISPC 1.14.0 release), but passes with LLVM 11.0.0 (rc1).

**Comment by dbabokin** [comment]
@Vsevolod-Livinskij I think it make sense to start testing ISPC with LLVM 11 instead of LLVM 10.

**Comment by dbabokin** [comment]
Here's the fix for this problem:
llvm/llvm-project@edcfb47ff6d

**Comment by dbabokin** [comment]
Fixed by #1846 and #1847
